ros2topic/ros2topic/verb/bw.py

```diff
@@ -62,19 +64,21 @@ class BwVerb(VerbExtension):
 
     def add_arguments(self, parser, cli_name):
         arg = parser.add_argument(
-            'topic',
+            'topic_name',
             help='Topic name to monitor for bandwidth utilization')
         arg.completer = TopicNameCompleter(
             include_hidden_topics_key='include_hidden_topics')
+        add_qos_arguments(parser, 'subscribe', 'sensor_data')
         parser.add_argument(
-            '--window', '-w', type=positive_int, default=DEFAULT_WINDOW_SIZE,
+            '--window', '-w', dest='window_size', type=positive_int, default=DEFAULT_WINDOW_SIZE,
             help='maximum window size, in # of messages, for calculating rate '
                  f'(default: {DEFAULT_WINDOW_SIZE})', metavar='WINDOW')
         add_direct_node_arguments(parser)
 
     def main(self, *, args):
+        qos_args = extract_qos_arguments(args)
         with DirectNode(args) as node:
-            _rostopic_bw(node.node, args.topic, window_size=args.window)
+            _rostopic_bw(node.node, args.topic_name, qos_args, window_size=args.window_size)
```

Comment:
i think we can call `choose_qos` before `_rostopic_bw` and pass the QoS profile argument to it. internal functions do not need to take the whole arguments, but only QoS profiles. i think that this formalizes the internal function better.